CHANGES.md

```diff
@@ -15,7 +15,29 @@
 [Unreleased](https://github.com/bird-house/birdhouse-deploy/tree/master) (latest)
 ------------------------------------------------------------------------------------------------------------------
 
-[//]: # (list changes here, using '-' for each new entry, remove this when items are added)
+## Changes
+
+- JupyterHub: update `jupyterhub` component default version to [5.2.1](https://github.com/Ouranosinc/jupyterhub/releases/tag/5.2.1-20241114)
+
+  This implements all changes between JupyterHub version 
+  [4.1.6 and 5.2.1](https://jupyterhub.readthedocs.io/en/stable/reference/changelog.html).
+
+  This update requires the following manual upgrade steps:
+
+  - If your local environment file sets the `c.DockerSpawner.image_whitelist` config option in the 
+    `JUPYTERHUB_ENABLE_MULTI_NOTEBOOKS` environnment variable. Change `c.DockerSpawner.image_whitelist` 
+    to `c.DockerSpawner.allowed_images`.
+  - Update the jupyterhub database with the following command after you have started up the stack:
+    ```
+    birdhouse compose exec jupyterhub jupyterhub upgrade-db
+    ```
```

Comment:
Can't this be defined as a compose service `jupyterhub-db-migration` with `restart: "no"`?

Reply:
It can. But I don't think it should. We do not want this to run every time jupyterhub starts up. This database upgrade is described as ["not too disruptive"](https://jupyterhub.readthedocs.io/en/latest/howto/upgrading-v5.html#database-upgrades) but more involved upgrades in the future may break things. This is really something that we want to do manually.

Reply:
Fair enough.